### PR TITLE
Fix for retrieving download URL for higher versions

### PR DESCRIPTION
**Pull request by Navya-Teja-Otturu**
The URL to download SOPS binary has changed after v3.7.1. The download URL now also includes the platform architecture(e.g., amd64 or arch64). Hence, adding a fix for it.

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -21988,9 +21988,28 @@ async function install(version2, chmod) {
...
 }
+function isVersionGreaterThan371(version2) {
+  const versionParts = version2.split(".");
+  const major = parseInt(versionParts[0]);
+  const minor = parseInt(versionParts[1]);
+  const patch = parseInt(versionParts[2]);
+  if (major > 3 || major === 3 && minor > 7 || major === 3 && minor === 7 && patch > 1) {
+    return true;
+  }
+  return false;
```
Comment (Uzlopak) [inline dist/index.js:22011]:
```suggestion
  return (
    (major > 3) ||
    (major === 3 && minor > 7) ||
    (major === 3 && minor === 7 && patch > 1)
  )
```

File: dist/index.js [hunk 2]
```diff
@@ -21988,9 +21988,28 @@ async function install(version2, chmod) {
...
+    }
+    return `https://github.com/getsops/sops/releases/download/v${version2}/sops-v${version2}.${extension}.${arch}`;
+  }
   return `https://github.com/getsops/sops/releases/download/v${version2}/sops-v${version2}.${extension}`;
 }
+function isVersionGreaterThan371(version2) {
+  const versionParts = version2.split(".");
+  const major = parseInt(versionParts[0]);
+  const minor = parseInt(versionParts[1]);
+  const patch = parseInt(versionParts[2]);
```
Comment (Uzlopak) [inline dist/index.js:22007]: can be simplified afaik
```suggestion
  const [major, minor, patch] = version2.split(".").map(v => parseInt(v, 10));
```

File: dist/index.js [hunk 3]
```diff
@@ -21988,9 +21988,28 @@ async function install(version2, chmod) {
   return binaryPath;
 }
 function downloadURL(version2) {
+  version2 = version2.startsWith("v") ? version2.slice(1) : version2;
+  const isHigherVersion = isVersionGreaterThan371(version2);
   const extension = process.platform === "win32" ? "exe" : process.platform;
+  if (isHigherVersion && extension !== "exe") {
+    let arch = process.arch.toString();
```
Comment (Uzlopak) [inline dist/index.js:21995]: is there a case where process.arch is not a string?

**Comment by uwehdaub** [comment]
LGTM

**Review by Uzlopak — APPROVED** [review]
LGTM

**Pull request merged**